src/App.tsx

```diff
@@ -18,30 +16,23 @@ const i18nInstance = initializeI18n({
 });
 
 const App = () => {
-    const [ShouldRedirectToDerivApp, isGBLoaded] = useGrowthbookGetFeatureValue({
-        featureFlag: 'redirect_to_deriv_app_p2p',
-    });
-    const [isOAuth2Enabled] = useOAuth2Enabled();
     const { onRenderAuthCheck } = useOAuth();
     const { init: initTrackJS } = useTrackjs();
     const { initialise: initDatadog } = useDatadog();
     const { isDesktop } = useDevice();
     const { initialise: initDerivAnalytics } = useDerivAnalytics();
     const isCallbackPage = getCurrentRoute() === 'callback';
+    const isProduction = process.env.VITE_NODE_ENV === 'production' || origin === URLConstants.derivP2pProduction;
+    const isStaging = process.env.VITE_NODE_ENV === 'staging' || origin === URLConstants.derivP2pStaging;
+    const isOAuth2Enabled = isProduction || isStaging;
 
     initTrackJS();
     initDerivAnalytics();
     initDatadog();
 
     useEffect(() => {
-        if (isGBLoaded && ShouldRedirectToDerivApp) {
-            const NODE_ENV = process.env.VITE_NODE_ENV;
-            const APP_URL = NODE_ENV === 'production' ? URLConstants.derivAppProduction : URLConstants.derivAppStaging;
-            window.location.href = `${APP_URL}/cashier/p2p`;
-        } else if (isGBLoaded) {
-            onRenderAuthCheck();
-        }
-    }, [isGBLoaded, ShouldRedirectToDerivApp, onRenderAuthCheck]);
```

Comment:
this GB flag was previously used when we were releasing p2p to production for certain countries. Its already out in production to all countries so theres no need to redirect to deriv app